packages/typegpu/src/data/struct.ts

```diff
@@ -121,13 +127,16 @@ class TgpuStructImpl<TProps extends Record<string, AnyTgpuData>>
   }
 }
 
-function getAttribute(field: AnyTgpuData): string | undefined {
-  if (field instanceof TgpuAlignedImpl) {
+function getAttribute<T extends AnyTgpuData>(field: T): string | undefined {
+  if (isAlignedSchema(field as unknown)) {
     return `@align(${field.byteAlignment}) `;
   }
-  if (field instanceof TgpuSizedImpl) {
+
+  if (isSizedSchema(field as unknown)) {
```

Comment:
why are these `as unknown` necessary?

Reply:
For some reason, TypeScript assumes after each of these `if` statements, that the `field` property can be nothing else that that condition, so it becomes `never`. I will debug it and fix it in another PR when I figure out what is wrong.